pipeline/beam_tables.py

```diff
@@ -310,76 +310,103 @@ def _parse_received_headers(headers: Dict[str, List[str]]) -> List[str]:
 
-def _flatten_measurement(filename: str, line: str) -> Iterator[Row]:
-  """Flatten a measurement string into several roundtrip rows.
+  def _process_non_satellite(self, filename: str, scan: Any,
+                             random_measurement_id: str) -> Iterator[Row]:
+    """Process a line of Echo/Discard/HTTP/S data.
 
-  Args:
-    filename: a filepath string
-    line: a json string describing a censored planet measurement. example
-    {'Keyword': 'test.com',
-     'Server': '1.2.3.4',
-     'Results': [{'Success': true},
-                 {'Success': false}]}
+    Args:
+      filename: a filepath string
+      scan: a loaded json object containing the parsed content of the line
+      random_measurement_id: a hex id identifying this individual measurement
 
-  Yields:
-    Dicts containing individual roundtrip information
-    {'column_name': field_value}
-    examples:
-    {'domain': 'test.com', 'ip': '1.2.3.4', 'success': true}
-    {'domain': 'test.com', 'ip': '1.2.3.4', 'success': false}
-  """
-  # pylint: disable=too-many-branches
-  try:
-    scan = json.loads(line)
-  except json.decoder.JSONDecodeError as e:
-    logging.warning('JSONDecodeError: %s\nFilename: %s\n%s\n', e, filename,
-                    line)
-    return
+    Yields:
+      Rows
+    """
+    for result in scan.get('Results', []):
+      date = result['StartTime'][:10]
+      row = {
+          'domain': scan['Keyword'],
+          'ip': scan['Server'],
+          'date': date,
+          'start_time': result['StartTime'],
+          'end_time': result['EndTime'],
+          'retries': scan['Retries'],
+          'sent': result['Sent'],
+          'blocked': scan['Blocked'],
+          'success': result['Success'],
+          'fail_sanity': scan['FailSanity'],
+          'stateful_block': scan['StatefulBlock'],
+          'measurement_id': random_measurement_id,
+          'source': _source_from_filename(filename),
+      }
 
-  blockpage_matcher = BlockpageMatcher()
+      if 'Received' in result:
+        received = result.get('Received', '')
+        received_fields = self._parse_received_data(received, scan['Blocked'])
+        row.update(received_fields)
 
-  # Add a unique id per-measurement so single retry rows can be reassembled
-  random_measurement_id = uuid.uuid4().hex
+      if 'Error' in result:
+        row['error'] = result['Error']
+
+      yield row
+
+  def _process_satellite(  # pylint: disable=no-self-use
+      self, filename: str, scan: Any,
+      random_measurement_id: str) -> Iterator[Row]:
+    """Process a line of Satellite data.
 
-  if 'Satellite' in filename:
+    Args:
+      filename: a filepath string
+      scan: a loaded json object containing the parsed content of the line
+      random_measurement_id: a hex id identifying this individual measurement
+
+    Yields:
+      Rows
+    """
     date = re.findall(r'\d\d\d\d-\d\d-\d\d', filename)[0]
     if date < "2021-03":
```

Comment:
This kind of thing can make the code hard to follow. I'd have 3 methods:
- formatVersion(filename) or isV1(filename)
- parseV1Line(line)
- parsev2Line(line)

Reply:
done